### DISCUSSION TITLE
Postal does not start after server restart

### DISCUSSION THREAD

**Discussion by rPawel** [postalserver/postal#1610] [Help with using Postal]
I have followed the https://docs.postalserver.io/install/installation/ instruction to get the postal up and running on a vanilla Ubuntu 20.04.3 VPS. It all went smoothly.
However once the server was restarted, the UI was showing white screen. I did docker ps and only rabbitmq, caddy and mariadb containers were running.
After that, I have manually invoked postal start --scale worker=8
so the workers started but the UI was still showing white.
I have then started postal_runner_run_xxxx container and then finally the UI was fine.
Is the above issue a bug or a lack of autostart configuration for these containers?

**Comment by boindil** [comment]
I worked around this by utilizing a systemd service:

```
[Unit]
Description=Postal Service
Requires=docker.service
After=docker.service

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/bin/postal start
ExecStop=/usr/bin/postal stop
TimeoutStartSec=0

[Install]
WantedBy=multi-user.target
```

**Comment by rPawel** [comment]
It does the job! Thanks @boindil !
I have set the command to `ExecStart=/usr/bin/postal start --scale worker=8`. One thing that I have noticed is that with the server restart the worker count is inaccurate.
However, when I start it with systemctl the worker count is as expected.

**Reply by willpower232** [reply]
You would need to see `postal status` when the UI isn't working to confirm which parts hadn't loaded properly. 

I don't think the scale setting persists if you don't add it to the docker-compose.yml file.

**Reply by rPawel** [reply]
ok so regarding the white screen and containers not starting: it seems that after the default installation, postal containers don't have a restart policy set.

docker inspect from postal_worker_5: 
```
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
```
but docker inspect from caddy has:
```
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
```

**Reply by rPawel** [reply]
Regarding using the @boindil workaround, to make sure the workers are up, I have just added some delay to postal start: 
`ExecStart=sh -c "sleep 10; /usr/bin/postal start --scale worker=8"`.
Silly, I know, but it works for now.

I guess postal has some logs which contain the failure reason.

**Reply by rPawel** [reply]
And I want to say, this idea for creating such software is just great. Thank you, guys! 🎉

**Reply by willpower232** [reply]
Ah that is interesting, I thought the restart policy was set somewhere but maybe I'm not remembering correctly.

The delay will be necessary as RabbitMQ particularly is slow to start so the delay roughly guarantees it will work.

**Reply by rPawel** [reply]
I guess for docker-compose world, it would be nice to utilise healthcheck on rabbitmq, e.g:
```
healthcheck:
  test: rabbitmq-diagnostics -q ping
  interval: 30s
  timeout: 30s
  retries: 3
```
and then use depends_on: https://docs.docker.com/compose/startup-order/

**Comment by pjv** [comment]
Here’s one potential docker-compose-based alternative to using a systemd service: https://github.com/postalserver/postal/discussions/1637#discussioncomment-1455003
